File: BOJ/10845/a.py
import sys

# with open('input.txt', 'r') as sys.stdin:
N = int(input())
Q = []

for i in range(0, N):
    S = str(input())
    split = S.split(' ')
    command = split[0]
    parameter = None
    if len(split) == 2:
        parameter = split[1]
    if command == "push":
        Q.append(parameter)
    elif command == "pop":
        if len(Q) != 0:
            print(Q[0])
            Q.pop(0)
        else:
            print(-1)
    elif command == "size":
        print(len(Q))
    elif command == "empty":
        if len(Q) == 0:
            print(1)
        else:
            print(0)
    elif command == "front":
        if len(Q) == 0:
            print(-1)
        else:
            print(Q[0])
    elif command == "back":
        if len(Q) == 0:
            print(-1)
        else:
            print(Q[-1])

File: BOJ/10845/a.cpp
#include <iostream>
#include <queue>
using namespace std;

int main()
{
    // freopen("input.txt", "rt", stdin);
    ios::sync_with_stdio(false);
    cin.tie(0);
    int N;
    cin >> N;
    queue<int> Q;
    for (int i = 0; i < N; i++)
    {
        string s;
        cin >> s;
        if (s == "push")
        {
            int num;
            cin >> num;
            Q.push(num);
        }
        else if (s == "pop")
        {
            if (!Q.empty())
            {
                int num = Q.front();
                cout << num << endl;
                Q.pop();
            }
            else
            {
                cout << -1 << endl;
            }
        }
        else if (s == "size")
        {
            cout << Q.size() << endl;
        }
        else if (s == "empty")
        {
            if (Q.empty())
            {
                cout << 1 << endl;
            }
            else
            {
                cout << 0 << endl;
            }
        }
        else if (s == "front")
        {
            if (!Q.empty())
            {
                int num = Q.front();
                cout << num << endl;
            }
            else
            {
                cout << -1 << endl;
            }
        }
        else if (s == "back")
        {
            if (!Q.empty())
            {
                int num = Q.back();
                cout << num << endl;
            }
            else
            {
                cout << -1 << endl;
            }
        }
    }
}